dask/highlevelgraph.py

```diff
@@ -317,161 +239,6 @@ def clone_value(o):
-        #   These can only be found in the known_key_dependencies cache, since
-        #   any dependencies computed in this method would have already had the
-        #   aliases removed.
-        if future_aliases:
-            alias_keys = set(future_aliases)
-            dependencies = {k: v - alias_keys for k, v in dependencies.items()}
-        # - Add in deps for any missing keys
-        missing_keys = dsk.keys() - dependencies.keys()
-
-        dependencies.update(
-            (k, keys_in_tasks(all_hlg_keys, [dsk[k]], as_list=False))
-            for k in missing_keys
-        )
-        # - Add in deps for any tasks that depend on futures
-        for k, futures in fut_deps.items():
-            if futures:
-                d = ensure_set(dependencies[k], copy=True)
-                d.update(f.key for f in futures)
-                dependencies[k] = d
-
-        # The scheduler expect all keys to be strings
-        dependencies = {
-            stringify(k): {stringify(dep) for dep in deps}
-            for k, deps in dependencies.items()
-        }
-
-        merged_hlg_keys = all_hlg_keys | dsk.keys()
-        dsk = {
-            stringify(k): stringify(v, exclusive=merged_hlg_keys)
-            for k, v in dsk.items()
-        }
-        dsk = toolz.valmap(dumps_task, dsk)
-        return {"dsk": dsk, "dependencies": dependencies}
-
-    @classmethod
-    def __dask_distributed_unpack__(
-        cls,
-        state: Any,
-        dsk: Mapping[str, Any],
-        dependencies: Mapping[str, set],
-    ) -> dict:
-        """Unpack the state of a layer previously packed by __dask_distributed_pack__()
-
-        This method is called by the scheduler in Distributed in order to unpack
-        the state of a layer and merge it into its global task graph. The method
-        can use `dsk` and `dependencies`, which are the already materialized
-        state of the preceding layers in the high level graph. The layers of the
-        high level graph are unpacked in topological order.
-
-        See Layer.__dask_distributed_pack__() for packing detail.
-
-        Parameters
-        ----------
-        state: Any
-            The state returned by Layer.__dask_distributed_pack__()
-        dsk: Mapping, read-only
-            The materialized low level graph of the already unpacked layers
-        dependencies: Mapping, read-only
-            The dependencies of each key in `dsk`
-
-        Returns
-        -------
-        unpacked-layer: dict
-            layer_dsk: Mapping[str, Any]
-                Materialized (stringified) graph of the layer
-            layer_deps: Mapping[str, set]
-                Dependencies of each key in `layer_dsk`
-        """
-        return {"dsk": state["dsk"], "deps": state["dependencies"]}
-
-    def __reduce__(self):
-        """Default serialization implementation, which materializes the Layer"""
-        return (MaterializedLayer, (dict(self),))
```

Comment:
There is actually no need to have any custom reducers. The old behavior was to materialize all layers by default when pickled. That's clearly not what we're after